gwcs/coordinate_frames.py

```diff
@@ -399,14 +429,29 @@ class TemporalFrame(CoordinateFrame):
     """
 
     def __init__(self, axes_order=(0,), reference_time=None,
-                 reference_frame=astropy.time.Time, unit=None,
+                 reference_frame=None, unit=None,
                  axes_names=None, name=None, axis_physical_types=None):
+        if reference_frame is None:
+            reference_frame = {}
+        elif not isinstance(reference_frame, dict):
+            raise TypeError("reference_frame must be a dict.")
 
         super().__init__(naxes=1, axes_type="TIME", axes_order=axes_order,
                          axes_names=axes_names, reference_frame=reference_frame,
                          unit=unit, name=name,
                          reference_position=reference_time, axis_physical_types=axis_physical_types)
 
+    @property
+    def _world_axis_object_classes(self):
+        return {'temporal': (
+            self.coordinates,
```

Comment:
@astrofrog @nden 

I have hacked this by putting a method of the class in here to do the conversion to `Time`. What it does is adds the time delta (returned as a float or quantity) from the model to the reference_time specified in the `TemporalFrame` constructor and returns a `Time` object.

This is really easy (and works) but I don't know if it massively violates the intent of APE14 for this to specify the *type* of the return. 

It would be useful to have a way to specify a callable here, and maybe use function annotations to describe what the return type of the callable is, and have this be a valid way of working for APE14?

Reply:
What about making a class factory here, such as ``TimeCoord``, which can take the reference time and then returns a ``Time`` object upon initialization? This would be compatible with APE 14.

Reply:
That at the end of the day is still a callable which returns a type different to the one that's in the dict. The fact that it is in itself a class seems irrelevant to all the introspection stuff you might want to do?

Reply:
TimeCoord could be a subclass of Time though? (that seems the safest if people try and check the type of the classes - that would be better than allowing arbitrary callables.)

Reply:
Yeah I suppose that fits all the criteria. Really feels like ramming a square peg in a round hole though.